ui/src/main/java/org/pentaho/di/ui/i18n/MessagesSourceCrawler.java

```diff
@@ -56,15 +39,93 @@
 import org.w3c.dom.Node;
 import org.w3c.dom.NodeList;
 
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 /**
  * This class takes care of crawling through the source code
  *
  * @author matt
- *
  */
 public class MessagesSourceCrawler {
 
-  private String[] scanPhrases;
+  /**
+   * When searching for a scanPhrase, it may exist over more than one source line. This string contains the characters
+   * to be considered in the calculation of the different splits that each scanPhrase may have.
+   *
+   * @see #SPLIT_CHARACTERS
+   * @see #SPLITTER_AND_WHITESPACES_PATTERN
+   */
+  private static final String SPLIT_CHARACTERS_STRING = ".(,";
+
+  /**
+   * When searching for a scanPhrase, it may exist over more than one source line. These are the characters to be
+   * considered in the calculation of the different splits that each scanPhrase may have.
+   *
+   * @see #SPLIT_CHARACTERS_STRING
+   * @see #SPLITTER_AND_WHITESPACES_PATTERN
+   */
+  private static final char[] SPLIT_CHARACTERS = SPLIT_CHARACTERS_STRING.toCharArray();
```

Comment:
The translator will process every line of every file it finds under the configured folder(s) and this can be a tight loop!
For every line it will see if it matches any of the scan phrases it was set to recognize and, for each of those, it has to recognize splitted phrases - and that's why we need to define which split characters (where a new line can appear) there are.
Before my changes, these (and others) strings were hard-coded and so, they were being created and destroyed every time with the associated costs on memory and GC... what I did here was create some single reusable static objects.
As we need the split characters in two distinct cases as String and as Character array, I had to differentiate them somehow (and thus the '_STRING'). The Character array object is declared based on the String object so that there's only one place to change if another split character is to be considered.